### PR TITLE
feat(37): Add bind social to profile

### PR DESCRIPTION
**Pull request by hromogen**
- Add functionality to bind a social account to user profile

- Refactor some code

### CODE REVIEW

File: packages/web/src/interfaces/app-state.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { RouterState } from 'connected-react-router';
+import { AuthState } from './auth-state';
+
+export interface AppState {
+  router: RouterState;
+  auth: AuthState;
+}
```
Comment (dubbha) [inline packages/web/src/interfaces/app-state.ts]: These belong in the store, why did suddenly decide to move them out to interfaces?
Makes no sense to separate the store from the interfaces describing the substores. Is this refactoring make things any better?

Comment (hromogen) [reply packages/web/src/interfaces/app-state.ts]: Well, the idea was to move all the interfaces into one folder to make them easier to track. I believe it to be more convenient but honestly there is no other solid reason.

File: packages/server/src/utils/create-tokenized-user.ts [hunk 2]
```diff
@@ -2,15 +2,40 @@ import UserEntity from '../models/user.entity';
 import UserWithToken from '../interfaces/tokenized.user.interface';
 
 import { createToken } from './create-token';
+import { Social } from 'constants/socials';
```
Comment (dubbha) [inline packages/server/src/utils/create-tokenized-user.ts]: eslint in server does not support this, so it fails here

File: packages/web/src/interfaces/app-state.ts (hunk 1 above)
Comment (dubbha) [reply packages/web/src/interfaces/app-state.ts]: I don't think it makes it easier to track, these interfaces have no meaning outside Redux, and everything Redux-related is co-located in the `store` directory, making it theretically possible to move and reuse together (even though it's not likely to happen). And inside `store` everything is groupped by features.

I agree though that it doesn't really belong in `actions`, so propose to move it to `reducers` instead, right above the initialState using it, and do the same for other substores in the future:

```ts
// store/auth/reducers.ts

export interface AuthState {
  isLoading: boolean;
  isAuthenticated: boolean;
  id: number | null;
  firstName: string | null;
  lastName: string | null;
  email: string | null;
  error: string | null;
  info: string | null;
  socialProfile: SocialProfile | null;
}

export const initialState: AuthState = {
  isAuthenticated: false,
  id: null,
  firstName: null,
  lastName: null,
  email: null,
  isLoading: false,
  error: null,
  info: null
  info: null,
  socialProfile: null
};
```

I believe the main `AppState` should stay in `store/index.ts` though, unless you can propose a better place inside `store`.

File: packages/web/src/interfaces/social-profile-provider.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export enum SocialProfileProvider {
+  FACEBOOK = 'Facebook',
+  GOOGLE = 'Google'
+}
```
Comment (dubbha) [inline packages/web/src/interfaces/social-profile-provider.ts]: enums belong in contants, not interfaces, if you're trying to keep everything well-organized

File: packages/web/src/interfaces/app-state.ts (hunk 1 above)
Comment (hromogen) [reply packages/web/src/interfaces/app-state.ts]: I don't think anything complex should live in the `index.ts` file. And for sure no file from the `store/` folder should import from it, that does not seem right at all

Comment (dubbha) [reply packages/web/src/interfaces/app-state.ts]: Agree with the `store/index.ts` part.

**Review by dubbha — APPROVED** [review]

**Comment by dubbha** [comment]
Refactoring is nice (though wasn't required), but the main feature functionality is working weirdly from the user point of view, at least for me (with the latest DB):
 - **Case #1:** Login with email/pass, you see the add facebook/google buttons, you press add facebook, both buttons disappear but nothing else changes in the UI. Then logout/login again, the buttons are there again.
 - **Case #2:** Login with facebook instead. You still see add facebook/google buttons, which is weird.

Both cases in a screencast:
rheei8Q5Rp

**Review by dubbha — CHANGES_REQUESTED** [review]
Please make the interaction meaningful for the user.

**Comment by hromogen** [comment]
> Please make the interaction meaningful for the user.

Done

File: packages/web/src/components/ManageSocials/AddSocialToProfile.tsx [hunk 4]
```diff
@@ -22,7 +22,9 @@ import {
 import './styles.scss';
 
 export const AddSocialToProfile = () => {
-  const { socialProfile } = useSelector(getAuth);
+  const auth = useSelector(getAuth);
+  console.log('_____AUTH_____', auth);
```
Comment (dubbha) [inline packages/web/src/components/ManageSocials/AddSocialToProfile.tsx]: Don't forget to remove this one in your next commit

**Review by dubbha — APPROVED** [review]

**Review by dartwlad — APPROVED** [review]

**Pull request merged**